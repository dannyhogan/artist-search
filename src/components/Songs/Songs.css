.Songs {

  list-style: none;
  background: transparent;
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  width: 820px;


  div {
    display: flex;
    align-items: center;
    width: 820px;
    height: 20px;
    background: rgba(76, 76, 76, 0.297);
    border-radius: 15px;
    border: 2px solid rgb(255, 255, 255);
    margin: 10px;
    transition: .5s;
    padding: 10px;
    color: white;

    img {
      width: 35px;
      margin-right: 10px;
    }
  }

  li {
    display: flex;
    align-items: center;
    width: 360px;
    background: rgba(76, 76, 76, 0.297);
    border-radius: 15px;
    border: 2px solid black;
    margin: 10px;
    transition: .5s;
    padding: 10px;

    a, a:visited {
      color: black;
    }

    img {
      width: 35px;
      margin-right: 10px;
    }
  }

  li:hover {
    background: rgba(255, 255, 255, 0.565);
    transition: .5s;
    transform: scale(1.05);
  }
}
