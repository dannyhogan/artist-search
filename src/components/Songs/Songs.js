import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from './Songs.css';
import note from '../../assets/note.png';
import i from '../../assets/i.png';

export default function Songs({ songs, name, release }) {

  const songList = songs.map(song => {
    return <li key={song.id}>
      <img src={note}></img>
      <Link to={`/song/${name}/${release}/${song.title}`}>{song.title}</Link>
    </li>;
  });

  return (
    <ul className={styles.Songs}>
      <div>
        <img src={i}></img>
        <p>Click on a song to view the lyrics!</p>
      </div>
      {songList}
    </ul>
  );
}

Songs.propTypes = {
  songs: PropTypes.array,
  name: PropTypes.string,
  release: PropTypes.string
};
