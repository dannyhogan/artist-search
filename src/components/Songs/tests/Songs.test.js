import React from 'react';
import { shallow } from 'enzyme';
import Songs from '../Songs';

describe('Songs component', () => {

  it('renders Songs component', () => {
    const songs = [{ title: 'Drake', id: '123' }];
    const wrapper = shallow(<Songs songs={songs} name="Drake" release="123" />);
    expect(wrapper).toMatchSnapshot();
  });
});
