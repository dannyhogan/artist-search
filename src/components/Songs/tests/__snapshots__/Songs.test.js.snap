// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Songs component renders Songs component 1`] = `
<ul
  className="Songs"
>
  <div>
    <img
      src="/path/image"
    />
    <p>
      Click on a song to view the lyrics!
    </p>
  </div>
  <li
    key="123"
  >
    <img
      src="/path/image"
    />
    <Link
      to="/song/Drake/123/Drake"
    >
      Drake
    </Link>
  </li>
</ul>
`;
